### PR TITLE
Initial support - TKC Candybar R3

### PR DESCRIPTION
**Pull request by TerryMathews**
## Description

Candybar R3 PCBs are using ATmega32u4 due to the supply constraint around STM32. The same switch matrix has been maintained so the source is largely unchanged aside from implementing the correct MCU defines and row and column pins. Same VID and PID are used so that it will utilize the same VIA definitions. This has been tested on a preproduction sample and works as intended.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tkc/candybar/lefty_r3/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
```
Comment (drashna) [inline keyboards/tkc/candybar/lefty_r3/keymaps/default/keymap.c:20]:
```suggestion
enum custom_layers {
    _BL,
    _FL,
};
```

File: keyboards/tkc/candybar/lefty_r3/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+![Candybar](https://cdn.shopify.com/s/files/1/1679/2319/products/CBR3CollectionHERO_720x.jpg)
+
+The Key Company Candybar is a staggered 40% board with a numpad utilizing the ATmega32u4 microcontroller.
+
+* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
+* Hardware Supported: TKC Candybar
+* Hardware Availability: [TheKey.Company](https://thekey.company/products/candybar-round-3)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/tkc/candybar/lefty_r3/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tkc/candybar/lefty_r3/readme.md (hunk 2 above)
Comment (TerryMathews) [reply keyboards/tkc/candybar/lefty_r3/readme.md:18]: Added to both readme files.

File: keyboards/tkc/candybar/lefty_r3/config.h [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/lefty_r3/config.h:102]:
```suggestion
```

File: keyboards/tkc/candybar/lefty_r3/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
+#define _AL 2
+#define _LL 3
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/lefty_r3/keymaps/via/keymap.c:22]:
```suggestion
enum layer_names {
    _BL, 
    _FL, 
    _AL, 
    _LL, 
};
```

File: keyboards/tkc/candybar/lefty_r3/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+BACKLIGHT_ENABLE = no
+BOOTMAGIC_ENABLE = yes  # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no # Mouse keys
+EXTRAKEY_ENABLE = yes # Audio control and System control
+CONSOLE_ENABLE = no  # Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes     # USB Nkey Rollover
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/lefty_r3/rules.mk:21]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

File: keyboards/tkc/candybar/righty_r3/config.h [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/righty_r3/config.h:102]:
```suggestion
```

File: keyboards/tkc/candybar/righty_r3/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/righty_r3/keymaps/default/keymap.c:20]:
```suggestion
enum layer_names {
    _BL, 
    _FL, 
};
```

File: keyboards/tkc/candybar/righty_r3/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
+#define _AL 2
+#define _LL 3
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/righty_r3/keymaps/via/keymap.c:22]:
```suggestion
enum layer_names {
    _BL, 
    _FL, 
    _AL, 
    _LL, 
};
```

File: keyboards/tkc/candybar/righty_r3/rules.mk [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+BACKLIGHT_ENABLE = no
+BOOTMAGIC_ENABLE = yes  # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no # Mouse keys
+EXTRAKEY_ENABLE = yes # Audio control and System control
+CONSOLE_ENABLE = no  # Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes     # USB Nkey Rollover
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (noroadsleft) [inline keyboards/tkc/candybar/righty_r3/rules.mk:21]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Just some codebase clean-up; no logic changes.

GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/15374/files).

**Review by drashna — APPROVED** [review]

**Pull request merged**